src/ucm/mmap/replace.c

```diff
@@ -16,19 +16,21 @@
 #include <ucs/sys/compiler.h>
 #include <ucs/sys/preprocessor.h>
 #include <ucs/type/component.h>
-#include <dlfcn.h>
+#include <pthread.h>
 
 
 #define MAP_FAILED ((void*)-1)
 
-static __thread int ucm_mmap_reentry = 0;
+static pthread_mutex_t ucm_mmap_get_orig_lock = PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP;
```

Comment:
because thread local storage implementation called malloc(), which resulted in a recursive invocation of sbrk()